thorlcr/activities/trace/thtraceslave.cpp

```diff
@@ -0,0 +1,159 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "thtraceslave.ipp"
+
+class CTraceSlaveActivity : public CSlaveActivity, public CThorDataLink, public CThorSteppable
+{
+    IThorDataLink *input;
+    IHThorTraceArg *helper;
+    IOutputMetaData *meta;
+    IRangeCompare *stepCompare;
```

Comment:
The CSteppable base class's member is set when the graph is wired, so this member variable and the setting of it, is not needed, I'll remove in a separate PR.